lib/features/run_placed/presentation/ui/screen/run_placed_screen.dart

```diff
@@ -1,58 +1,77 @@
 import 'dart:convert';
 
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:hotfoot/features/navigation_screen/presentation/bloc/navigation_screen_bloc.dart';
 import 'package:hotfoot/features/run_placed/presentation/ui/widgets/accept_delivery_button.dart';
 import 'package:hotfoot/features/run_placed/presentation/ui/widgets/cancel_delivery_button.dart';
 import 'package:hotfoot/features/run_placed/presentation/ui/widgets/open_close_chat_button.dart';
+import 'package:hotfoot/features/runs/domain/use_cases/get_run_stream.dart';
+import 'package:hotfoot/injection_container.dart';
 
 class RunPlacedScreen extends StatelessWidget {
+
+  Stream<QuerySnapshot> returnCurrentRunStream (GetRunStream useCase, String id) async* {
+    final result = await useCase.call(id);

```

Comment:
The call function is special. It makes the object callable. So you can just do useCase(id);

Reply:
You can also use the `fold` method on an either. Do a fold and return `null` for the left case. You can return what is returned from a fold like this:

```dart
Stream<QuerySnapshot> yourFunction() {
    final runStreamEither = await getRunStream();
    return runStreamEither.fold(
        (failure) => null,
        (runStream)  => runStream,
   );
}

```

Reply:
Avoids having to create another async generator (`async*`) function.

Reply:
Did you mean ?
```
Stream<QuerySnapshot> yourFunction() async {
```
Await keyword only works in async function... Also, assuming that you meant await I am getting a compile error =>
```
Stream<QuerySnapshot> returnCurrentRunStream (GetRunStream useCase, String id) async {
    final runStreamEither = await useCase(id);
    return runStreamEither.fold(
      (failure) => null,
      (runStream) => runStream,
    );
  }
```
This is giving me the error ( A value of type 'FutureOr<Stream<QuerySnapshot>> can't be returned from method 'returnCurrentRunStream' because it has a return type of 'Stream<QuerySnapshot>'. )

Reply:
assuming that you meant async*

Reply:
You're right. I see the problem  now. Let me just change the useCase to not return an either. Since we don't have our won bloc here it's a little tricky to await something in the UI's build fuction.

Reply:
Actually let's go with what you had in the beginning. Sorry about that. I didn't see the problem before.